src/box/box.cc

```diff
@@ -3621,7 +3653,17 @@ local_recovery(const struct tt_uuid *instance_uuid,
 		struct replica *master;
 		if (replicaset_needs_rejoin(&master)) {
 			say_crit("replica is too old, initiating rebootstrap");
-			return bootstrap_from_master(master);
+			while (true) {
+				do {
+					box_restart_replication();
```

Comment:
I'd rewrite this a bit. Seems to me it's possible that `replicaset_needs_rejoin` returns `false` after one or more iterations of `box_restart_replication()`.

For example, the instance couldn't connect to the one remote node which holds the WALS for it.
Then the instance fails `bootstrap_from_master`, and reconnects to all the nodes.

I'm trying to say you need something similar to:

```c
while(replicaset_needs_rejoin(&master) {
    if (bootstrap_from_master(master) == 0)
         return;
    box_restart_replication()
}
```

Reply:
I understand like this:
If the current instance is too far behind its peers in the replica set, it must be rebooted. `replicaset_needs_rejoin()` returns true and sets the master to the instance that will be used to rejoin.

Since the master has been installed, there is no need to look for a new master after reconnecting to all nodes.

Reply:
My point was that it's possible that the master which holds the needed WALs might be unreachable. That's why on some iteration `replicaset_needs_rejoin` returns true.

Now after some attempts the master might become reachable again. In this case `replicaset_needs_rejoin` will return false, and the instance will continue without rejoining.

No changes in code required. Your current version is correct.